cms/djangoapps/models/settings/course_details.py

```diff
@@ -188,6 +192,12 @@ def update_from_json(cls, course_key, jsondict, user):
             descriptor.language = jsondict['language']
             dirty = True
 
+        if (SelfPacedConfiguration.current().enabled
+                and 'self_paced' in jsondict
+                and jsondict['self_paced'] != descriptor.self_paced):
+            descriptor.self_paced = jsondict['self_paced']
+            dirty = True
```

Comment:
Does this need to be conditional? Also, should we set dirty only if the value of `self_paced` has actually changed?